src/plots/cartesian/dragbox.js

```diff
@@ -178,6 +178,8 @@ function makeDragBox(gd, plotinfo, x, y, w, h, ns, ew) {
             prepSelect(e, startX, startY, dragOptions, dragModeNow);
         } else {
             dragOptions.clickFn = clickFn;
+            // clear selection polygon cache (if any)
+            plotinfo.selection = false;
```

Comment:
Should this be part of `clearSelect`? That gets called in all 3 blocks below (directly in the first and third, and in the second it's inside `zoomPrep` - looks like these can be simplified anyway) but also in [`zoomWheel`](https://github.com/plotly/plotly.js/blob/7784a10880545ef79b5dbc269d5450f337c6807d/src/plots/cartesian/dragbox.js#L397) - does that case have the same bug?

Reply:
Ah. Good catch. Zoom has the problem. 

Working on a fix :pick:

Reply:
fixed and :lock: in https://github.com/plotly/plotly.js/pull/2676/commits/f682fb8979da56083c4a513fe7cbe642f323a2e3